hal/src/rtl872x/usart_hal.cpp

```diff
@@ -326,15 +327,18 @@ class Usart {
     }
 
     ssize_t flush() {
-        startTransmission();
+        CHECK_TRUE(isEnabled(), SYSTEM_ERROR_INVALID_STATE);
         while (true) {
+            while (transmitting_) {
+                HAL_Delay_Milliseconds(5);
+                // FIXME: busy loop
```

Comment:
What's the ultimate fix for this?

Reply:
We still need to figure it out, but for now we at least have a delay inside of this loop, which allows lower priority tasks to run.

Reply:
Might be a good idea to retest on top of #2581